Userland/Libraries/LibCore/MimeData.cpp

```diff
@@ -54,167 +54,129 @@ void MimeData::set_text(DeprecatedString const& text)
     set_data("text/plain", text.to_byte_buffer());
 }
 
-static AK::Array<StringView, 8> s_plaintext_suffixes = {
+// FIXME: Share this, TextEditor and HackStudio language detection somehow.
+static Vector s_plaintext_suffixes = {
+    // Extensions
     ".c"sv,
     ".cpp"sv,
     ".gml"sv,
     ".h"sv,
     ".hpp"sv,
     ".ini"sv,
     ".ipc"sv,
-    ".txt"sv
-};
+    ".txt"sv,
 
-static AK::Array<StringView, 3> s_plaintext_basenames = {
+    // Base names
     ".history"sv,
     ".shellrc"sv
     "CMakeLists.txt"sv,
 };
 
-// FIXME: Share this, TextEditor and HackStudio language detection somehow.
-static bool should_contain_plain_text_based_on_filename(StringView path)
-{
-    for (auto suffix : s_plaintext_suffixes) {
-        if (path.ends_with(suffix))
-            return true;
-    }
-    return s_plaintext_basenames.contains_slow(LexicalPath::basename(path));
-}
+static Array s_registered_mime_type = {
```

Comment:
Again probably should not be mutable data?

Reply:
Oh and it can't be `constexpr`  because placement `new` isn't `constexpr`, and required by `Optional`

Reply:
I know, but this one can be `const` (I didn't say `constexpr` for this one) :)